src/Routes/Default/Default.tsx

```diff
@@ -7,29 +7,60 @@ import { notificationsAtom, useRemoveNotification } from '../../state/notificati
 import Header from '../../Components/Header/Header';
 import React, { useEffect } from 'react';
 import useCurrentUser from '../../hooks/useCurrentUser';
-import { LayoutTypes, getWidgetMapping } from '../../api/dashboard-templates';
+import { LayoutTypes, WidgetPermission, getWidgetMapping } from '../../api/dashboard-templates';
 import { widgetMappingAtom } from '../../state/widgetMappingAtom';
 import '../../App.scss';
 import Portal from '@redhat-cloud-services/frontend-components-notifications/Portal';
+import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
 
 const DefaultRoute = (props: { layoutType?: LayoutTypes }) => {
   const isLayoutLocked = useAtomValue(lockedLayoutAtom);
   const notifications = useAtomValue(notificationsAtom);
   const removeNotification = useRemoveNotification();
   const setWidgetMapping = useSetAtom(widgetMappingAtom);
   const { currentUser } = useCurrentUser();
+  const { visibilityFunctions } = useChrome();
+
+  const checkPermissions = async (permissions: WidgetPermission[]): Promise<boolean> => {
+    console.log('Checking permissions: ', permissions);
+    for (const permission of permissions) {
+      if (permission.method === 'permissions' && permission.args && !(await visibilityFunctions.hasPermissions(permission.args))) {
+        return false;
+      }
+
+      if (permission.method === 'loosePermissions' && permission.args && !(await visibilityFunctions.loosePermissions(permission.args))) {
+        return false;
+      }
+
+      if (permission.method === 'isOrgAdmin' && !(await visibilityFunctions.isOrgAdmin())) {
+        return false;
+      }
+    }
+
+    return true;
```

Comment:
This for loop only allows 3 methods. We want all of those available in chrome. This PR exports the type https://github.com/RedHatInsights/frontend-components/pull/1999 Also, it might be better to use `every`, without it we'd only check for one permission.
```JS
const resolved = Promise.all(permissions.map(async ({ method, args }) => (method && await (visibilityFunctions[method] as any))(...(args || [])))).every(Boolean)
```